frontend/app/routes/protected/person-case/birth-details.tsx

```diff
@@ -0,0 +1,275 @@
+
+      context.session.inPersonSINCase ??= {};
+      context.session.inPersonSINCase.birthDetails = parseResult.output;
+
+      throw i18nRedirect('routes/protected/request.tsx', request);
+    }
+
+    default: {
+      throw new AppError(`Unrecognized action: ${action}`, ErrorCodes.UNRECOGNIZED_ACTION);
+    }
+  }
+}
+
+export default function BirthDetails({ loaderData, actionData, params }: Route.ComponentProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const [country, setCountry] = useState<string | undefined>(loaderData.defaultFormValues.firstName);
+  const [fromMultiple, setFromMultiple] = useState<boolean | undefined>(
+    loaderData.defaultFormValues.supportingDocumentsRequired,
+  );
+  const fetcherKey = useId();
+  const fetcher = useFetcher<Info['actionData']>({ key: fetcherKey });
+  const isSubmitting = fetcher.state !== 'idle';
+  const errors = fetcher.data?.errors;
+  const canadaCountryCode = 'CAN';
+
+  function handleCountryChange(event: ChangeEvent<HTMLSelectElement>) {
+    setCountry(event.target.value);
+  }
+
+  const countryOptions = ['select-option', ...COUNTRIES].map((value) => ({
+    value: value === 'select-option' ? '' : value,
+    children: t(`protected:birth-details.countries.${value}` as 'protected:birth-details.countries.select-option'),
+  }));
+
+  function handleFromMultipleChanged(event: ChangeEvent<HTMLInputElement>) {
+    setFromMultiple(event.target.value === REQUIRE_OPTIONS.yes);
+  }
+
+  const fromMultipleOptions: InputRadiosProps['options'] = [
+    {
+      children: t('gcweb:input-option.yes'),
+      value: REQUIRE_OPTIONS.yes,
+      defaultChecked: fromMultiple === true,
+      onChange: handleFromMultipleChanged,
+    },
+    {
+      children: t('gcweb:input-option.no'),
+      value: REQUIRE_OPTIONS.no,
+      defaultChecked: fromMultiple === false,
+      onChange: handleFromMultipleChanged,
+    },
+  ];
+
+  return (
+    <>
+      <div className="flex justify-end">
+        <Button id="abandon-button" endIcon={faXmark} variant="link">
+          {t('protected:person-case.abandon-button')}
+        </Button>
+        <Button id="refer-button" endIcon={faExclamationCircle} variant="link">
+          {t('protected:person-case.refer-button')}
+        </Button>
+      </div>
+      <Progress className="mt-8" label="" value={30} />
+      <PageTitle subTitle={t('protected:in-person.title')}>{t('protected:birth-details.page-title')}</PageTitle>
+      <FetcherErrorSummary fetcherKey={fetcherKey}>
+        <fetcher.Form method="post" noValidate>
+          <div className="space-y-6">
+            <InputSelect
+              id="country-id"
+              name="country"
+              errorMessage={errors?.country?.at(0)}
+              defaultValue={loaderData.defaultFormValues.firstName ?? ''}
+              required
+              options={countryOptions}
+              label={t('protected:birth-details.country.label')}
+              onChange={handleCountryChange}
```

Comment:
Instead of defining a whole new function for this, you should be able to just do something like this:

``` tsx
onChange={({ target }) => setCountry(target.value)}
```